### ISSUE TITLE
Configuration Tool Needed

### ISSUE THREAD

**Issue by skaller** [felix-lang/felix#152]
Felix needs a configuration tool. I keep trying to write one and fail.

The main purpose of the tool is to set up new target directories. This involves:

1. Set up the target config directory. This depends on only three parameters:
* the compiler
* the machine word size 32/64
* the OS
and is a simple mechanical copy of *.fpc files from the tree structured src/config.

2. set the standard felix macros flx.flxh to the right values for the OS

3. Munge any C++ headers. There is only one now I think.

Build and install the target. This can be tricky. On Linux, we might be using a gcc cross compiler toolchain to build for Windows, or for Android. On MacOS we might be building for iPhone. On Windows there are a huge number of options, including building under Ubuntu for MSVC, cygwin64, using minGW, and others. The reality is in some cases Felix can run to generate C++ but it can't be compiled until shipped to another platform. In some cases a cross compiler is available but we can't link. In some cases we can link. In some cases we can even run the binary, eg iPhoneSimulator can run emulations of iPhone programs on MacOS. Cygwin, we can run. Etc. Controlling what stage is what and even running parts on one OS and parts on another are possible from a single tool. Some people even have a Mac with Parallels to run Windows.

In addition, one would like a file in .felix/config that says what targets to build. This raises another problem. At present, for host, the bootstrap sets up a configuration, and then clobbers it with overwrites from .felix/config. But this only works for host target. The whole cache is also for a single target, you cannot do concurrent builds for two targets. If you build for one target and switch to another, the cache is cleared and rebuilt from scratch each time. 

In summary, Felix has much better facilities than any other system for handling multiple targets, but at present in-depth knowledge is required to configure them, and it has to be done by hand, the painful way. Even some tools to automate parts of this would be good.

I personally would like to build on MacOS for both clang and gcc to compare performance.

**Comment by arabelladonna** [comment]
I just wanted to weigh in here and say that I think that trying to support every environment is overkill. Some thoughts:

* Cygwin is a last resort. Binaries that aren't compiled _exactly right_ within Cygwin permanently rely on its runtime. This is generally used by developers who absolutely want to have a Windows version without doing any of the work to port their software to Windows. I've even had situations where I've compiled compilers (that transpile to C, mind) in Cygwin and had them automatically inject code that has a Cygwin dependency when transpiling. It's implementation-specific, but it's not a reliable solution for native executables (it is way too much work to configure properly) and Cygwin's runtime has pretty significant performance overhead.
* MSYS2 is generally _the_ bash environment for creating native Windows executables. If you want to build a native executable with `make` and `gcc`, this is the way. This is probably the best way to create as close to a platform-independent solution as you can. _However_, this also creates a dependence on that environment and that has positives and negatives.
* MSVC's workflow goes without saying. That's how the system currently builds on Windows.

If you want my honest opinion, you should pick one workflow for Windows and just stick with it. The C++ code generated is intended to be platform-independent. Those who want to use an environment like Cygwin or MSYS2 could fairly easily work that into their build process once the executables are built. IIRC, there are already flags that determine which C/C++ compiler suite is used.
